src/main/groovy/org/pih/warehouse/outbound/ImportPackingListItem.groovy

```diff
@@ -28,11 +28,12 @@ class ImportPackingListItem implements Validateable {
         if (source['lotNumber']) {
             return source['lotNumber']
         }
+        ProductAvailabilityService productAvailabilityService = Holders.grailsApplication.mainContext.getBean(ProductAvailabilityService)
 
         // otherwise infer inventory item based on provided bin-location
-        Location internalLocation = Location.findByNameAndParentLocation(source['binLocation'], obj.origin)
         Product product = Product.findByProductCode(source['product'])
-        List<ProductAvailability> items = ProductAvailability.findAllByProductAndBinLocationAndLocation(product, internalLocation, obj.origin)
+        List<ProductAvailability> items = productAvailabilityService.getAvailableBinLocations(obj.origin, product?.productCode)
+                .findAll { it?.binLocation?.name == source['binLocation'] }
```

Comment:
since there is a lot of those methods in the product availability service, I won't force that, but I would prefer to have this filtering in the query itself, instead of calling `.findAll` just after doing the fetch.
Please take a look if there is a method that would return product availability for pair: `origin`, `product`, `binLocation` and use it if possible.
Otherwise, if the performance is OK for an item, that has stock in a lot of bins, I'm fine with that, not to implement another comprehensive method in that service.

Reply:
I don't think we have a one right now that does a search for specific bin locaitons

Reply:
yeah, we don't and I would need to implement a new service method for that.
I don't think we are dealing with too much data at this point and maybe a simple `findAll` search should be enough

Reply:
ok, I guess we can just monitor and live with it for now, but I wanted to make a point, that this usually should not be the way to go.